extensions/ql-vscode/src/extension.ts

```diff
@@ -467,9 +468,14 @@ async function activateWithInstalledDistribution(
     selectedQuery: Uri | undefined,
     progress: ProgressCallback,
     token: CancellationToken,
+    databaseQuickPick: DatabaseItem | undefined,
   ): Promise<void> {
     if (qs !== undefined) {
-      const dbItem = await databaseUI.getDatabaseItem(progress, token);
+      const dbItem = databaseQuickPick !== undefined
+        // database selected from multi-database quick pick
+        ? databaseQuickPick
+        // database currently selected in Databases UI
+        : await databaseUI.getDatabaseItem(progress, token);
```

Comment:
If you take the above suggestion, this code becomes a little simpler.
```suggestion
     // If no database is specified, use the database currently selected in the Databases UI
     dbItem = dbItem || await databaseUI.getDatabaseItem(progress, token);
```